### ISSUE TITLE
Segmentation Fault on Exit

### ISSUE THREAD

**Issue by David-Estevez** [asrob-uc3m/robotDevastation#8]
Segmentation fault on exit. Traceback:

```
0 std::_Rb_tree<int, std::pair<int const, rd::RdPlayer>, std::_Select1st<std::pair<int const, rd::RdPlayer> >, std::less<int>, std::allocator<std::pair<int const, rd::RdPlayer> > >::begin stl_tree.h  685 0x444334
1 std::map<int, rd::RdPlayer, std::less<int>, std::allocator<std::pair<int const, rd::RdPlayer> > >::begin  stl_map.h   321 0x443464
2 rd::RdMentalMap::getPlayers   RdMentalMap.cpp 55  0x441cb9
3 rd::RobotDevastation::updateModule    RobotDevastation.cpp    223 0x43cfc3
4 yarp::os::RFModule::runModule()   /usr/local/lib/libYARP_OS.so.1      0x7ff66da66312
5 main  main.cpp    91  0x43a41a    
```

**Comment by jgvictores** [comment]
Occurs even with --mockupRobotManager --mockupImageManager

**Comment by David-Estevez** [comment]
Different trace this time:
```
1	nanosleep	syscall-template.S	81	0x7ffff5d3cf3d	
2	yarp::os::impl::sleepThread(ACE_Time_Value&)			0x7ffff7204f4c	
3	yarp::os::RFModule::runModule()			0x7ffff7205363	
4	main	main.cpp	91	0x44d605	
```
On main thread, and:
```
1	sem_wait	sem_wait.S	44	0x7ffff4946650	
2	RateThreadCallbackAdapter::run()			0x7ffff71f50b1	
3	theExecutiveBranch(void *)			0x7ffff723b125	
4	ACE_OS_Thread_Adapter::invoke()			0x7ffff3437b16	
5	start_thread	pthread_create.c	312	0x7ffff4940182	
6	clone	clone.S	111	0x7ffff5d7647d	
```
On the other thread

**Comment by David-Estevez** [comment]
Last SEGFAULT is solved now (07b52ba), we were displaying a value in the main thread using a mentalMap pointer that was never checked, so when the user exits the game, the pointer becomed NULL and the program crashed. It still crashes, but it is only related to the first and second traces on the previous comment.

**Comment by jgvictores** [comment]
Closing, has not happened in a long time at `develop`.
